mex/extractors/international_projects/transform.py

```diff
@@ -70,15 +82,35 @@ def transform_international_projects_source_to_extracted_activity(
 
     all_partner_organizations: list[MergedOrganizationIdentifier] = []
     if source.partner_organization:
-        all_partner_organizations.extend(
-            wfc
-            for fc in source.partner_organization
-            if (wfc := partner_organizations_stable_target_id_by_query.get(fc))
-        )
+        for partner_org in source.partner_organization:
+            if wpo := partner_organizations_stable_target_id_by_query.get(partner_org):
+                all_partner_organizations.append(wpo)
+            else:
+                extracted_organization = ExtractedOrganization(
+                    officialName=[Text(value=partner_org)],
+                    identifierInPrimarySource=partner_org,
+                    hadPrimarySource=extracted_primary_source.stableTargetId,
+                )
+                load([extracted_organization])
+                all_partner_organizations.append(
+                    MergedOrganizationIdentifier(extracted_organization.stableTargetId)
+                )
 
+    activity_types = international_projects_activity["activityType"][0]["mappingRules"]
+    activity_types_dict: dict[str, ActivityType] = {}
+    for activity_type in activity_types:
+        for_value = activity_type.get("forValues")
+        set_value = activity_type.get("setValues")[0]
+        if not for_value:
+            activity_types_dict.setdefault("Other", set_value)
+            continue
+        activity_types_dict.setdefault(for_value[0], set_value)
```

Comment:
```suggestion
    if (
        source.funding_type
        == international_projects_activity["activityType"][0]["mappingRules"][0][
            "forValues"
        ][0]
    ):
        activity_type = international_projects_activity["activityType"][0][
            "mappingRules"
        ][0]["setValues"][0]
    elif (
        source.funding_type
        == international_projects_activity["activityType"][0]["mappingRules"][1][
            "forValues"
        ][0]
    ):
        activity_type = international_projects_activity["activityType"][0][
            "mappingRules"
        ][1]["setValues"][0]
    else:
        activity_type = international_projects_activity["activityType"][0][
            "mappingRules"
        ][2]["setValues"][0]
```
I think, this would be a much simpler logic for a simpler logic for one single string with three differenct cases

Reply:
how about this? efe7130042b2651d61c47e790665299f03383bda

Reply:
use `activity_type_from_mapping` in line 102 as well

Reply:
done in 423d407d1e16506941b9208b13c27e5f97eef58b